### ISSUE TITLE
History of sensors no longer updating after backup 

### ISSUE THREAD

**Issue by silviudc** [sabeechen/hassio-google-drive-backup#957]
Having a weird issue with this add-on, I have it set to do daily backups to google drive and some days after the backup is done, none of my sensors are recording their states, I have to restart HA to get them working again.

Can be seen last night, the straight line, and then recovered after HA restart:
![image](https://github.com/sabeechen/hassio-google-drive-backup/assets/17740184/48691c0a-c72c-4be1-925b-ffd6e86c7b71)

add-on version: 0.112.1
HA: 
![image](https://github.com/sabeechen/hassio-google-drive-backup/assets/17740184/cfab8dc6-ae18-4ec7-9405-1a6943d5b7f3)

Nothing useful in the logs:
`12-09 20:00:00 INFO Syncing Backups
12-09 20:00:00 INFO Requesting a new backup
12-09 21:08:47 INFO A backup directory file was modified, we'll check for new backups soon.
12-09 21:12:27 INFO A backup directory file was modified, we'll check for new backups soon.
12-09 21:12:27 INFO Backup finished
12-09 21:12:27 INFO Syncing Backups
12-09 21:12:29 INFO Deleting 'Full Backup 2023-12-02 20:00:00' from Home Assistant
12-09 21:12:29 INFO Uploading 'Full Backup 2023-12-09 20:00:00' to Google Drive
12-09 21:16:23 INFO Deleting 'Full Backup 2023-12-05 20:00:00' From Google Drive
12-09 21:16:24 INFO Syncing Backups
12-09 21:19:28 INFO A backup directory file was modified, we'll check for new backups soon.
12-09 22:47:09 INFO Syncing Backups
12-10 00:41:57 INFO Syncing Backups
12-10 03:39:50 INFO Syncing Backups
12-10 05:13:57 INFO Syncing Backups
12-10 07:06:54 INFO Syncing Backups
12-10 08:41:07 INFO Syncing Backups
12-10 10:18:48 INFO Syncing Backups`

And not seeing any errors in HA logs either around that time.

**Comment by sabeechen** [comment]
A couple things come to mind:
 - If you use an addon for your database (MariaDB, Influx, etc) make sure the database is running.
 - If you stop addons while taking a backup, try turning that off.

If there is somethign wrong with your database, that'll usually show up in 1 of two places:
1. The Home Assistant Logs: "Settings" > "System > "Logs"
2. The Supervisor logs.  These are in the same place, but you need to fiddle with some Hoem Assistant settings to make it visible. [See this FAQ item](https://github.com/sabeechen/hassio-google-drive-backup#im-getting-weird-errors--where-do-i-look-for-more-detils-about-an-error-supervisor-logs) for how to get there.

**Comment by silviudc** [comment]
Yes using MariaDB and I'm not turning anything off while doing the backups but this add-on must be doing that to the DB while the backup is running right? 
Just seems to not be able to turn it back on after the backup is done (sometimes)
Is there a method I could force the DB to come back online via an automation perhaps?
Nothing out of the ordinary in either log

**Comment by sabeechen** [comment]
This addon can be configured to stop addons while backing up, though I don't recommend it.  Just make sure you don't have the "Stop Addons" section enabled in your addon settings.  If MariaDB is left running during a backup (which is normal) HA will put it into a state that buffers writes to the database while the backup is in progress.  I suspect that something goes wrong after the backup when HA brings it out of that state.  Can you check what MariaDB's logs look like after a backup finishes?

**Comment by silviudc** [comment]
So this was last night where everything worked fine and the issue did not happen:
```
s6-rc: info: service mariadb-lock-core: starting
[20:00:06] INFO: Start MariaDB client (to lock tables for backups)
s6-rc: info: service mariadb-lock-core successfully started
s6-rc: info: service mariadb-lock-post: starting
[20:00:06] INFO: MariaDB tables locked
s6-rc: info: service mariadb-lock-post successfully started
s6-rc: info: service mariadb-lock-post: stopping
[20:58:56] INFO: MariaDB tables unlocked
s6-rc: info: service mariadb-lock-post successfully stopped
s6-rc: info: service mariadb-lock-core: stopping
[20:58:56] INFO: MariaDB client (to lock tables for backups) stopped
s6-rc: info: service mariadb-lock-core successfully stopped
s6-rc: info: service mariadb-lock-core: starting
[20:00:06] INFO: Start MariaDB client (to lock tables for backups)
s6-rc: info: service mariadb-lock-core successfully started
s6-rc: info: service mariadb-lock-post: starting
[20:00:06] INFO: MariaDB tables locked
s6-rc: info: service mariadb-lock-post successfully started
s6-rc: info: service mariadb-lock-post: stopping
[21:08:23] INFO: MariaDB tables unlocked
s6-rc: info: service mariadb-lock-post successfully stopped
s6-rc: info: service mariadb-lock-core: stopping
[21:08:23] INFO: MariaDB client (to lock tables for backups) stopped
s6-rc: info: service mariadb-lock-core successfully stopped
s6-rc: info: service mariadb-lock-core: starting
[20:00:06] INFO: Start MariaDB client (to lock tables for backups)
s6-rc: info: service mariadb-lock-core successfully started
s6-rc: info: service mariadb-lock-post: starting
[20:00:06] INFO: MariaDB tables locked
s6-rc: info: service mariadb-lock-post successfully started
s6-rc: info: service mariadb-lock-post: stopping
[20:51:26] INFO: MariaDB tables unlocked
s6-rc: info: service mariadb-lock-post successfully stopped
s6-rc: info: service mariadb-lock-core: stopping
[20:51:27] INFO: MariaDB client (to lock tables for backups) stopped
s6-rc: info: service mariadb-lock-core successfully stopped
s6-rc: info: service mariadb-lock-core: starting
[20:00:06] INFO: Start MariaDB client (to lock tables for backups)
s6-rc: info: service mariadb-lock-core successfully started
s6-rc: info: service mariadb-lock-post: starting
[20:00:06] INFO: MariaDB tables locked
s6-rc: info: service mariadb-lock-post successfully started
s6-rc: info: service mariadb-lock-post: stopping
[20:52:37] INFO: MariaDB tables unlocked
s6-rc: info: service mariadb-lock-post successfully stopped
s6-rc: info: service mariadb-lock-core: stopping
[20:52:37] INFO: MariaDB client (to lock tables for backups) stopped
s6-rc: info: service mariadb-lock-core successfully stopped
s6-rc: info: service mariadb-lock-core: starting
[20:00:05] INFO: Start MariaDB client (to lock tables for backups)
s6-rc: info: service mariadb-lock-core successfully started
s6-rc: info: service mariadb-lock-post: starting
[20:00:05] INFO: MariaDB tables locked
s6-rc: info: service mariadb-lock-post successfully started
s6-rc: info: service mariadb-lock-post: stopping
[20:56:21] INFO: MariaDB tables unlocked
s6-rc: info: service mariadb-lock-post successfully stopped
s6-rc: info: service mariadb-lock-core: stopping
[20:56:21] INFO: MariaDB client (to lock tables for backups) stopped
s6-rc: info: service mariadb-lock-core successfully stopped
```
Will keep an eye out tonight as it might happen

**Comment by silviudc** [comment]
Did it again today, nothing out of the ordinary in the mariaDB logs
```
s6-rc: info: service mariadb-lock-core: starting
[20:00:06] INFO: Start MariaDB client (to lock tables for backups)
s6-rc: info: service mariadb-lock-core successfully started
s6-rc: info: service mariadb-lock-post: starting
[20:00:06] INFO: MariaDB tables locked
s6-rc: info: service mariadb-lock-post successfully started
s6-rc: info: service mariadb-lock-post: stopping
[20:58:46] INFO: MariaDB tables unlocked
s6-rc: info: service mariadb-lock-post successfully stopped
s6-rc: info: service mariadb-lock-core: stopping
[20:58:46] INFO: MariaDB client (to lock tables for backups) stopped
s6-rc: info: service mariadb-lock-core successfully stopped
s6-rc: info: service mariadb-lock-core: starting
[20:00:05] INFO: Start MariaDB client (to lock tables for backups)
s6-rc: info: service mariadb-lock-core successfully started
s6-rc: info: service mariadb-lock-post: starting
[20:00:06] INFO: MariaDB tables locked
s6-rc: info: service mariadb-lock-post successfully started
s6-rc: info: service mariadb-lock-post: stopping
[20:57:31] INFO: MariaDB tables unlocked
s6-rc: info: service mariadb-lock-post successfully stopped
s6-rc: info: service mariadb-lock-core: stopping
[20:57:31] INFO: MariaDB client (to lock tables for backups) stopped
s6-rc: info: service mariadb-lock-core successfully stopped
```

**Comment by sabeechen** [comment]
Well thats interesting. It looks like if the recorder backlog gets too high it just gives up forever.  I'd expect it to be a little more clever if it can tell the backlog is high because the database is being backed up.  This might be worth creating an issue in Home Assistant, from the looks of it this isn't a problem specific to this addon and would affect anyone with a setup like yours even if they backup through Home Assistant's UI.

It doesn't look like HA's recorder lets you configure the size of the backlog.  I'm not sure of a good solution for this but some thoughts are:
 - Make those sensor record less often.
 - Keep a shorter history so backups happen faster.
 - [Exclude the frequently written entities ](https://www.home-assistant.io/integrations/recorder/#configure-filter)from being saved.
 - Call the [recorder.enable](https://www.home-assistant.io/integrations/recorder/#service-enable) service some time after a backup.  I'm not sure if it will recover from a "failure" like this though.

But none of them are good solutions.  I'd consider this a bug in HA, though I can't say if the HA devs would also see it that way.

**Comment by silviudc** [comment]
I agree, now that I have seen that error on the HA log I think this is just HA behaviour and not really this add-on so I will create an issue there with the same.
I have recorder set to save 7 days so in my mind that's not that high.
These sensors are important so not logging them at all is not really an option, need to look into their update intervals. 
I tried calling recorder.enable when it was "stopped" and it did not recover, had to reboot HA
Maybe hardware is limitation, it's a 2GB rpi4 with SATA SSD, hoping can upgrade to something with nvme and maybe then it will backup quicker?
